File: contains_duplicate_ii.py
import random
class Solution:
    # @param {integer[]} nums
    # @param {integer} k
    # @return {boolean}
    def containsNearbyDuplicate(self, nums, k):
        dic = {}
        n = len(nums)
        count = 0
        for i in range(n):
            if nums[i] not in dic:
                dic[nums[i]] = i
            else:
                if i - dic[nums[i]] <= k:
                    count +=1
        return count == 1
                
        
if __name__=='__main__':
    sol=Solution()
    nums = [random.randint(1, 10) for i in range(10)]
    k = random.randint(1, 10)
    print nums, k
    print sol.containsNearbyDuplicate(nums,k)

File: Algorithm/CPP/contains_duplicate_ii.cpp
# include <stdio.h>
# include <iostream>
# include <vector>
# include <algorithm>
using namespace std;


class Solution {
public:
	bool containsNearbyDuplicate(vector<int>& nums, int k) {
		unordered_map<int, int> myMap;
		int count = 0;
		for (int i = 0; i < nums.size();i++)
		{
			if (myMap.find(nums[i])==myMap.end())
			{
				myMap[nums[i]] = i;
			}
			else
			{
				if (i - myMap[nums[i]] <= k)
				{
					count++;
				}
				myMap[nums[i]] = i;
			}
		}
	
		return count == 1;
	}
};

int main()
{
	Solution sol;
	vector<int> nums = {1,0,1,1};
	string s = "ccc";
	string t = "cc";
	cout << sol.containsNearbyDuplicate(nums,1);

	cout << endl;
	//for (auto k :nums )cout << k << " ";
	system("pause");
	return 0;
}
